plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/ProbabilisticSampler.java

```diff
@@ -67,7 +109,7 @@ public SamplingResult shouldSample(
         Attributes attributes,
         List<LinkData> parentLinks
     ) {
-        return getSampler().shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
+        return getSampler(attributes).shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
```

Comment:
Action based sampler should only be used if the give span is a ROOT span otherwise the defaultSampler would apply.

Reply:
Done!